extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetSchemaReader.java

```diff
@@ -166,10 +187,28 @@ public static ParquetInstructions readParquetSchema(
                 return column.getPath();
             });
             final String colName;
-            final String mappedName = readInstructions.getColumnNameFromParquetColumnName(parquetColumnName);
-            if (mappedName != null) {
-                colName = mappedName;
-            } else {
+            COL_NAME: {
+                FIELD_ID: if (fieldId != null) {
+                    if (fieldIdCount.getOrDefault(fieldId.intValue(), 0L) > 1) {
+                        // This file has multiple entries for fieldId; don't match against it for field ids.
+                        break FIELD_ID;
+                    }
+                    final List<String> columnNames =
+                            readInstructions.getColumnNamesFromParquetFieldId(fieldId.intValue());
+                    if (columnNames.size() == 1) {
+                        colName = columnNames.get(0);
+                        break COL_NAME;
+                    } else if (columnNames.size() > 1) {
+                        throw new IllegalArgumentException(String.format(
```

Comment:
I think this limitation is entirely because you didn't want to refactor the code. If you're going to argue for that, we should at least guide the user to use an `updateView` to achieve their goals.

Reply:
Agreed. Added a comment that this could be improved with refactoring of the code.

Reply:
Did you also update the error message? If we're not going to let the user do this, tell them how they can achieve the same result.